src/impl/icetransport.cpp

```diff
@@ -154,6 +133,39 @@ IceTransport::IceTransport(const Configuration &config, candidate_callback candi
 	mAgent = decltype(mAgent)(juice_create(&jconfig), juice_destroy);
 	if (!mAgent)
 		throw std::runtime_error("Failed to create the ICE agent");
+
+	// Add TURN servers
+	for (auto &server : servers) {
+		if (mTURNServersAdded++ >= MAX_TURN_SERVERS_COUNT) {
+			break;
+		}
```

Comment:
The counter incorrectly counts STUN servers too, so you might end up with a single TURN server or none added if you set STUN servers. It should be done once in `addIceServer()` after ensuring the type is indeed TURN.

Reply:
Done! Not needed anymore because we filter.